streams/test-utils/src/test/java/org/apache/kafka/streams/MockProcessorContextTest.java

```diff
@@ -0,0 +1,410 @@
+            }
+        };
+
+        final MockProcessorContext context = new MockProcessorContext();
+
+        processor.init(context);
+
+        try {
+            processor.process("foo", 5L);
+            fail("Should have thrown an UnsupportedOperationException.");
+        } catch (final UnsupportedOperationException uoe) {
+            // expected
+        }
+    }
+
+    /**
+     * Behavioral test demonstrating the use of the context for capturing commits
+     */
+    @Test
+    public void shouldCaptureCommitsAndAllowReset() {
+        final AbstractProcessor<String, Long> processor = new AbstractProcessor<String, Long>() {
+            private int count = 0;
+
+            @Override
+            public void process(final String key, final Long value) {
+                if (++count > 2) context().commit();
+            }
+        };
+
+        final MockProcessorContext context = new MockProcessorContext();
+
+        processor.init(context);
+
+        processor.process("foo", 5L);
+        processor.process("barbaz", 50L);
+
+        Assert.assertFalse(context.committed());
+
+        processor.process("foobar", 500L);
+
+        Assert.assertTrue(context.committed());
+
+        context.resetCommit();
+
+        Assert.assertFalse(context.committed());
+    }
+
+    /**
+     * Behavioral test demonstrating the use of state stores
+     */
+    @Test
+    public void shouldStoreAndReturnStateStores() {
+        final AbstractProcessor<String, Long> processor = new AbstractProcessor<String, Long>() {
+            @Override
+            public void process(final String key, final Long value) {
+                //noinspection unchecked
+                final KeyValueStore<String, Long> stateStore = (KeyValueStore<String, Long>) context().getStateStore("my-state");
+                stateStore.put(key, (stateStore.get(key) == null ? 0 : stateStore.get(key)) + value);
+                stateStore.put("all", (stateStore.get("all") == null ? 0 : stateStore.get("all")) + value);
+            }
+        };
+
+        final MockProcessorContext context = new MockProcessorContext();
+        final KeyValueStore<String, Long> store = new InMemoryKeyValueStore<>("my-state", Serdes.String(), Serdes.Long());
+        context.register(store, false, null);
+
+        processor.init(context);
+
+        processor.process("foo", 5L);
+        processor.process("bar", 50L);
+
+        Assert.assertEquals(5L, (long) store.get("foo"));
+        Assert.assertEquals(50L, (long) store.get("bar"));
+        Assert.assertEquals(55L, (long) store.get("all"));
+    }
+
+    /**
+     * Behavioral test demonstrating the use of the context with context-aware processors
+     */
+    @Test
+    public void shouldCaptureApplicationAndRecordMetadata() {
+        final Properties config = new Properties();
+        config.put(StreamsConfig.APPLICATION_ID_CONFIG, "testMetadata");
+        config.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "");
```

Comment:
nit: remove -- not used

Reply:
Can you point out where it would fail? Unclear to me atm.

Reply:
ah, I commented on the wrong usage. See my stacktraces comment on the MockProcessorContext constructor.